src/decompression.jl

```diff
@@ -616,53 +632,98 @@ function decompress!(
-            else
-                # uplo = :U or uplo = :F
-                # A[i,j] is stored at index_ij = (A.colptr[j] + offset_U) in A.nzval
-                if uplo != :L
-                    nzind = A_colptr[j] + upper_triangle_offsets[counter]
-                    nzA[nzind] = val
+        # We need the buffer only when the tree is not a star (trivial or non-trivial)
+        if !is_star[k]
+            # Reset the buffer to zero for all vertices in a tree (except the root)
+            for (vertex, _) in reverse_bfs_orders[k]
+                buffer_right_type[vertex] = zero(R)
+            end
+            # Reset the buffer to zero for the root vertex
+            (_, root) = reverse_bfs_orders[k][end]
+            buffer_right_type[root] = zero(R)
+
+            # The tree is not a star
+            for (i, j) in reverse_bfs_orders[k]
+                counter += 1
+                val = B[i, color[j]] - buffer_right_type[i]
+                buffer_right_type[j] = buffer_right_type[j] + val
+
+                #! format: off
+                # A[i,j] is in the lower triangular part of A
+                if in_triangle(i, j, :L)
+                    # uplo = :L or uplo = :F
+                    # A[i,j] is stored at index_ij = (A.colptr[j+1] - offset_L) in A.nzval
+                    if uplo != :U
+                        nzind = A_colptr[j + 1] - lower_triangle_offsets[counter]
+                        nzA[nzind] = val
+                    end
+
+                    # uplo = :U or uplo = :F
+                    # A[j,i] is stored at index_ji = (A.colptr[i] + offset_U) in A.nzval
+                    if uplo != :L
+                        nzind = A_colptr[i] + upper_triangle_offsets[counter]
+                        nzA[nzind] = val
+                    end
+
+                # A[i,j] is in the upper triangular part of A
+                else
+                    # uplo = :U or uplo = :F
+                    # A[i,j] is stored at index_ij = (A.colptr[j] + offset_U) in A.nzval
+                    if uplo != :L
+                        nzind = A_colptr[j] + upper_triangle_offsets[counter]
+                        nzA[nzind] = val
+                    end
+
+                    # uplo = :L or uplo = :F
+                    # A[j,i] is stored at index_ji = (A.colptr[i+1] - offset_L) in A.nzval
+                    if uplo != :U
+                        nzind = A_colptr[i + 1] - lower_triangle_offsets[counter]
+                        nzA[nzind] = val
+                    end
                 end
-
-                # uplo = :L or uplo = :F
-                # A[j,i] is stored at index_ji = (A.colptr[i+1] - offset_L) in A.nzval
-                if uplo != :U
-                    nzind = A_colptr[i + 1] - lower_triangle_offsets[counter]
-                    nzA[nzind] = val
+                #! format: on
+            end
+        else
+            # The tree is a star (trivial or non-trivial)
```

Comment:
Can we reduce code duplication here?

Reply:
I think that we should define an auxiliary function for that

Reply:
I added a function `compute_tree_value` that determine the value of `A[i,j]` / `A[j,i]`.
It takes into account if the tree is a star or not with the help of multiple dispatch for a faster recovery.